pkg/component/observability/monitoring/kubestatemetrics/kubestatemetrics.go

```diff
@@ -174,6 +175,14 @@ func (k *kubeStateMetrics) Deploy(ctx context.Context) error {
 			return err
 		} else if err == nil && mr.Spec.Class == nil {
 			// spec.class nil means shoot
+
+			// remove the finalizer to prevent deleting the cluster role and cluster role binding in the shoot even when the GRM is not running
+			// in a hibernated control-plane. It is similar to "mr.Spec.KeepObjects = false", but it supports hibernated shoots as well.
```

Comment:
I guess you meant `keepObjects = true` since this is the case where the objects will not be deleted if the MR is deleted.

```suggestion
			// in a hibernated control-plane. It is similar to "mr.Spec.KeepObjects = true", but it supports hibernated shoots as well.
```

Reply:
Thanks!